web/pgadmin/settings/static/ApplicationStateProvider.jsx

```diff
@@ -63,36 +45,64 @@ export function ApplicationStateProvider({children, toolDataId}){
     return saveAppState;
   };
 
-  async function getQueryToolContent() {
+  async function getToolContent(transId) {
     try {
-      let transId = toolDataId.split('-')[1];
       const res = await getApiInstance({'Content-Encoding': 'gzip'}).get(
         url_for('settings.get_tool_data', {
           'trans_id': transId,
         })
       );
-      return res.data.success? JSON.parse(res.data.data.result.tool_data): null;
+
+      if (!res?.data?.success) {
+        console.warn('Unable to retrieve tool content.');
+        return null;
+      }
+      
+      const toolData = res.data.data.result;
+      const connectionInfo = toolData?.connection_info;
+      const toolContent = JSON.parse(toolData.tool_data);
+
+      let loadFile = false;
+      let fileName = null;
+
+      if(connectionInfo?.open_file_name){
+        fileName = connectionInfo.open_file_name;
+
+        if(connectionInfo.is_editor_dirty){
+          if(connectionInfo.external_file_changes){
+            // file has external chages
+            return {loadFile: loadFile, fileName: fileName, data: toolContent, modifiedExternally: true};
+          }
+        }else{
+          loadFile = true;
+          return {loadFile: loadFile, fileName: fileName, data: null};
+        }
+      }
+      return {loadFile: loadFile, fileName: fileName, data: toolContent};
+
     } catch (error) {
-      pgAdmin.Browser.notifier.pgRespErrorNotify(error);
+      let errorMsg = gettext(error?.response?.data?.errormsg || error);
+      console.warn(errorMsg);
+      pgAdmin.Browser.notifier.pgRespErrorNotify(errorMsg);
       return null;
     }
   }
 
-  const deleteToolData = ()=>{
-    let transId = toolDataId.split('-')[1];
-    getApiInstance().delete(
-      url_for('settings.delete_application_state'), {data:{'panelId': transId}}
-    ).then(()=> { /* Sonar Qube */}).catch(function(error) {
-      pgAdmin.Browser.notifier.pgRespErrorNotify(error);
-    });
-
+  const deleteToolData = (panelId)=>{
+    if (saveAppState && [BROWSER_PANELS.QUERY_TOOL, BROWSER_PANELS.ERD_TOOL, BROWSER_PANELS.SCHEMA_DIFF_TOOL, BROWSER_PANELS.PSQL_TOOL].includes(panelId.split('_')[0])){
```

Comment:
What is this check for? If I add new workspace then this also need to be changed which can be missed.